UnityProject/Assets/Scripts/UI/Action/ActionControlInventory.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using Mirror;
+
+public class ActionControlInventory : MonoBehaviour, IClientInventoryMove
+{
+	public ActionController ActionControllerType = ActionController.Inventory;
+
+	public List<IActionGUI> ControllingActions = new List<IActionGUI>();
+
+	public void OnInventoryMoveClient(ClientInventoryMove info)
+	{
+		var pna = PlayerManager.LocalPlayerScript.playerNetworkActions;
+		var showAlert = pna.GetActiveHandItem() == gameObject ||
+						pna.GetOffHandItem() == gameObject;
+		foreach (var _IActionGUI in ControllingActions)
+		{
+			UIActionManager.Instance.SetAction(_IActionGUI, showAlert);
+		}
+	}
+
+	void Start()
+	{
+		var ActionGUIs = this.GetComponents<IActionGUI>();
+		foreach (var ActionGUI in ActionGUIs) {
+			if (!ActionGUI.ActionData.PreventBeingControlledBy.Contains(ActionControllerType)) {
```

Comment:
so, Let's say if you have a action implemented on your Item E.G The spell Call to yourself, When you drop it you want to keep the spell because The spell is Summoning the item to you so it wouldn't make sense if it would disappear when you drop it, but however It has another Component for delinking The spell and That's getting turned off and on by the inventory script, so you wouldn't want the summons spell to be turned off  by the inventory Script

Reply:
This sounds like a complex way to solve that unique problem, and it's pulling this complexity into a very fundamental part of the action system - the ActionData, just to solve this unique problem. Can't this be solved without such a complicated field, such as a flag? Maybe a "Don't Remove When Dropped" field (something like that). PreventBeingControlledBy is hard to make sense of so I wouldn't want it to be part of the main API for working with actions.

Reply:
I don't feel like it's a complicated Thing, like it's going to be hardly used,
and it's just a list of stuff saying that you should ignore me if you're this